summariser/src/aggregator/holochain_metrics.rs

```diff
@@ -0,0 +1,228 @@
+use crate::analyze::{counter_stats, gauge_stats, standard_timing_stats};
+use crate::frame::LoadError;
+use crate::model::{
+    CounterStats, GaugeStats, HolochainDatabaseMetrics, HolochainMetrics, HolochainMetricsConfig,
+};
+use crate::model::{HolochainDatabaseKind, StandardTimingsStats, StrumVariantSelector};
+use crate::partition::partition_by_tags;
+use crate::query::query_metrics;
+use anyhow::Context;
+use std::collections::BTreeMap;
+use wind_tunnel_summary_model::RunSummary;
+
+/// Try to aggregate Holochain metrics.
+///
+/// If it fails returns [`None`] and reports the error in the logs.
+pub async fn try_aggregate_holochain_metrics(
+    client: &influxdb::Client,
+    summary: &RunSummary,
+    config: HolochainMetricsConfig,
+) -> Option<HolochainMetrics> {
+    match aggregate_holochain_metrics(client, summary, config).await {
+        Ok(metrics) => Some(metrics),
+        Err(e) => {
+            log::error!("Failed to aggregate holochain metrics: {e}");
+            None
+        }
+    }
+}
+
+async fn aggregate_holochain_metrics(
+    client: &influxdb::Client,
+    summary: &RunSummary,
+    config: HolochainMetricsConfig,
+) -> anyhow::Result<HolochainMetrics> {
+    log::debug!("Aggregating holochain metrics for run {}", summary.run_id);
+
+    Ok(HolochainMetrics {
+        p2p_request_duration: if !config.p2p_request_duration {
+            BTreeMap::new()
+        } else {
+            let res = aggregate_duration(
+                client,
+                summary,
+                "hc.holochain_p2p.request.duration.s",
+                &["dna_hash"],
+                None,
+            )
+            .await;
+            match res {
+                Ok(tree) => tree,
+                Err(e) => match e.downcast_ref::<LoadError>() {
+                    Some(LoadError::NoSeriesInResult { .. }) => BTreeMap::new(),
+                    None => return Err(e).context("Aggregate p2p_request_duration"),
```

Comment:
It is expected that some metrics might not be generated by a scenario. So instead of ending the whole process, we just set this metrics to `None` or to an empty `BTreeMap`.